tests/integration_tests/functional/test_new_tap.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Check that network is working after restore with a different tap."""
+import pytest
+from integration_tests.functional.test_cpu_features import PLATFORM
+from framework.artifacts import SnapshotType
+from framework.builder import SnapshotBuilder, MicrovmBuilder
+import framework.utils as utils
+import host_tools.network as net_tools
+
+
+@pytest.mark.skipif(
+    PLATFORM != "x86_64",
+    reason="There is no real need to test this on both platforms."
+)
+def test_snapshot_delete_tap(bin_cloner_path):
+    """Test that network works after restore with a different tap.
+
+    Creates a vm and initialize a network connection(i.e a ping) and create a
+    snapshot. After the snapshot is created delete the tap and restore the vm.
+    build_from_snapshot method will create a new tap for us. After restore
+    there should be no packets lost on first connection.
+    """
+    builder = MicrovmBuilder(bin_cloner_path)
+    vm_instance = builder.build_vm_nano()
+    microvm = vm_instance.vm
+
+    microvm.start()
+
+    ssh_connection = net_tools.SSHConnection(microvm.ssh_config)
+    _, _, _ = ssh_connection.execute_command("ping -c 4 192.168.0.1")
+    _, _, stderr = ssh_connection.execute_command("ip -s -s neigh flush all")
+    assert stderr.read() == ""
+    snapshot_builder = SnapshotBuilder(microvm)
+    disks = [vm_instance.disks[0].local_path()]
+    snapshot = snapshot_builder.create(disks,
+                                       vm_instance.ssh_key,
+                                       snapshot_type=SnapshotType.FULL)
+    utils.run_cmd("ip netns exec {} ip link del tap0"
+                  .format(microvm.jailer.netns))
+    microvm.kill()
+    vm = builder.build_from_snapshot(snapshot,
+                                     resume=True,
+                                     diff_snapshots=False)[0]
+    ssh_connection_restore = net_tools.SSHConnection(vm.ssh_config)
+    _, out, _ = ssh_connection_restore.execute_command("ping -c 4 192.168.0.1")
```

Comment:
IMO the test basically validates that after resuming from a snapshot, the guest network stack will have stale/invalid ARP cache entries that need flushing before egress traffic works. Is there any Firecracker regression are we defending against here ?

Reply:
Not really, deleted the integration test.